src/key.rs

```diff
@@ -650,6 +650,12 @@ impl KeyPair {
             Ok(())
         }
     }
+
+    /// Gets the [XOnlyPublicKey] for this [KeyPair].
+    #[inline]
+    pub fn public_key(&self) -> XOnlyPublicKey {
```

Comment:
@tcharding @apoelstra 
Post merge comment: The name is somewhat confusing because there is already a `PublicKey`(different from `XOnlyPublicKey`) type that the impl shown below. I know that we have had the discussion of return type in function names several times, but I think based on confusion between `XOnlyPublicKey` and `PublicKey`, we should have named to `x_only_public_key`. Perhaps we should add another method `public_key` that actually returns a full `PublicKey`.

Because this is already released code, we can perhaps use the names `pub_key` and `x_only_pub_key` ? What do you think?

```rust
impl From<KeyPair> for PublicKey {
    #[inline]
    fn from(pair: KeyPair) -> Self {
        PublicKey::from_keypair(&pair)
    }
}
```

Reply:
The deprecate thing is just to try and be kind to our users, right? No need to be dogmatic with it if we botch something up and there is no nice way to fix it nicely while using 'deprecated since'. Also trait impls can't be deprecated anyways so we cannot be dogmatic about it.

Reply:
Sorry, for the long message. I was talking about the first commit. Not the second one that deprecates another method. My basic objection was that `.public_key` should have returned the `PublicKey` type, but it returns the `XonlyPublicKey` type.

Reply:
oh my bad :)